tests/integration/states/user.py

```diff
@@ -25,25 +26,41 @@
 import salt.utils
 import integration
 
+if salt.utils.is_darwin():
+    USER = 'macuser'
+    GROUP = 'macuser'
+    GID = randint(400,500)
+    NOGROUPGID = randint(400,500)
+else:
+    USER = 'nobody'
+    GROUP = 'nobody'
+    GID = 'nobody'
+    NOGROUPGID = 'nogroup'
 
 class UserTest(integration.ModuleCase,
                integration.SaltReturnAssertsMixIn):
     '''
     test for user absent
     '''
+    def setUp(self):
+        if salt.utils.is_darwin():
+            #on mac we need to add user, because there is
+            #no creationtime for nobody user.
+            add_user = self.run_function('user.add', [USER], gid=GID)
+
     def test_user_absent(self):
         ret = self.run_state('user.absent', name='unpossible')
         self.assertSaltTrueReturn(ret)
 
     def test_user_if_present(self):
```

Comment:
I know you didn't originally write these tests, but these tests should be marked as destructive, and also need to be run as root like the other tests in this file. Can you add those to the 3-or-so tests that you've updated here?